server/src/test/java/org/elasticsearch/cluster/metadata/AutoExpandReplicasTests.java

```diff
@@ -216,79 +213,6 @@ public void testAutoExpandWhenNodeLeavesAndPossiblyRejoins() throws InterruptedE
-            ); // local node is the master
-            allNodes.add(localNode);
-            allNodes.add(oldNode);
-            ClusterState state = ClusterStateCreationUtils.state(
-                localNode,
-                localNode,
-                allNodes.toArray(new DiscoveryNode[0]),
-                TransportVersions.V_7_0_0
-            );
-
-            CreateIndexRequest request = new CreateIndexRequest(
-                "index",
-                Settings.builder().put(SETTING_NUMBER_OF_SHARDS, 1).put(SETTING_AUTO_EXPAND_REPLICAS, "0-all").build()
-            ).waitForActiveShards(ActiveShardCount.NONE);
-            state = cluster.createIndex(state, request);
-            assertTrue(state.metadata().hasIndex("index"));
-            while (state.routingTable().index("index").shard(0).allShardsStarted() == false) {
-                logger.info(state);
-                state = cluster.applyStartedShards(
-                    state,
-                    state.routingTable().index("index").shard(0).shardsWithState(ShardRoutingState.INITIALIZING)
-                );
-                state = cluster.reroute(state, new ClusterRerouteRequest());
-            }
-
-            DiscoveryNode newNode = createNode(Version.V_7_6_0, DiscoveryNodeRole.MASTER_ROLE, DiscoveryNodeRole.DATA_ROLE); // local node
-                                                                                                                             // is the
-                                                                                                                             // master
-
-            state = cluster.addNode(state, newNode, TransportVersions.V_7_6_0);
-
-            // use allocation filtering
-            state = cluster.updateSettings(
-                state,
-                new UpdateSettingsRequest("index").settings(
-                    Settings.builder().put(IndexMetadata.INDEX_ROUTING_EXCLUDE_GROUP_PREFIX + "._name", oldNode.getName()).build()
-                )
-            );
-
-            while (state.routingTable().index("index").shard(0).allShardsStarted() == false) {
-                logger.info(state);
-                state = cluster.applyStartedShards(
-                    state,
-                    state.routingTable().index("index").shard(0).shardsWithState(ShardRoutingState.INITIALIZING)
-                );
-                state = cluster.reroute(state, new ClusterRerouteRequest());
-            }
-
-            // check that presence of old node means that auto-expansion does not take allocation filtering into account
-            assertThat(state.routingTable().index("index").shard(0).size(), equalTo(3));
-
-            // remove old node and check that auto-expansion takes allocation filtering into account
-            state = cluster.removeNodes(state, Collections.singletonList(oldNode));
-            assertThat(state.routingTable().index("index").shard(0).size(), equalTo(2));
```

Comment:
I am not sure this test was correct :confused: 
* it is removing an old node, but there is another one with the same version
* it is removing node that is excluded